SU2_CFD/src/numerics/turbulent/turb_sources.cpp

```diff
@@ -816,88 +822,96 @@ CNumerics::ResidualType<> CSourcePieceWise_TurbSST::ComputeResidual(const CConfi
-   }
-   else {
-     pk = Eddy_Viscosity_i*StrainMag_i*StrainMag_i - 2.0/3.0*Density_i*TurbVar_i[0]*diverg;
-   }
-
-
-   pk = min(pk,20.0*beta_star*Density_i*TurbVar_i[1]*TurbVar_i[0]);
-   pk = max(pk,0.0);
-
-   zeta = max(TurbVar_i[1], VorticityMag*F2_i/a1);
-
-   /* if using UQ methodolgy, calculate production using perturbed Reynolds stress matrix */
-
-   if (using_uq){
-     pw = PerturbedStrainMag * PerturbedStrainMag - 2.0/3.0*zeta*diverg;
-   }
-   else {
-     pw = StrainMag_i*StrainMag_i - 2.0/3.0*zeta*diverg;
-   }
-   pw = alfa_blended*Density_i*max(pw,0.0);
-
-   /*--- Sustaining terms, if desired. Note that if the production terms are
-         larger equal than the sustaining terms, the original formulation is
-         obtained again. This is in contrast to the version in literature
-         where the sustaining terms are simply added. This latter approach could
-         lead to problems for very big values of the free-stream turbulence
-         intensity. ---*/
-
-   if ( sustaining_terms ) {
-     const su2double sust_k = beta_star*Density_i*kAmb*omegaAmb;
-     const su2double sust_w = beta_blended*Density_i*omegaAmb*omegaAmb;
-
-     pk = max(pk, sust_k);
-     pw = max(pw, sust_w);
-   }
-
-   /*--- Add the production terms to the residuals. ---*/
-
-   Residual[0] += pk*Volume;
-   Residual[1] += pw*Volume;
-
-   /*--- Dissipation ---*/
-
-   Residual[0] -= beta_star*Density_i*TurbVar_i[1]*TurbVar_i[0]*Volume;
-   Residual[1] -= beta_blended*Density_i*TurbVar_i[1]*TurbVar_i[1]*Volume;
-
-   /*--- Cross diffusion ---*/
-
-   Residual[1] += (1.0 - F1_i)*CDkw_i*Volume;
-
-   /*--- Implicit part ---*/
-
-   Jacobian_i[0][0] = -beta_star*TurbVar_i[1]*Volume;
-   Jacobian_i[0][1] = -beta_star*TurbVar_i[0]*Volume;
-   Jacobian_i[1][0] = 0.0;
-   Jacobian_i[1][1] = -2.0*beta_blended*TurbVar_i[1]*Volume;
+    
+    /*--- Production ---*/
+    
+    diverg = 0.0;
+    for (iDim = 0; iDim < nDim; iDim++)
+      diverg += PrimVar_Grad_i[iDim+1][iDim];
+    
+    /* if using UQ methodolgy, calculate production using perturbed Reynolds stress matrix */
+    
+    if (using_uq){
+      ComputePerturbedRSM(nDim, Eig_Val_Comp, uq_permute, uq_delta_b, uq_urlx,
+                          PrimVar_Grad_i+1, Density_i, Eddy_Viscosity_i,
+                          TurbVar_i[0], MeanPerturbedRSM);
+      SetPerturbedStrainMag(TurbVar_i[0]);
+      pk = Eddy_Viscosity_i*PerturbedStrainMag*PerturbedStrainMag
+           - TWO3*Density_i*TurbVar_i[0]*diverg;
+    }
+    else {
+      pk = Eddy_Viscosity_i*StrainMag_i*StrainMag_i - TWO3*Density_i*TurbVar_i[0]*diverg;
+    }
+    
+    pk = min(pk,20.0*beta_star*Density_i*TurbVar_i[1]*TurbVar_i[0]);
+    pk = max(pk,0.0);
+    
+    zeta = max(TurbVar_i[1], VorticityMag*F2_i/a1);
+    
+    /* if using UQ methodolgy, calculate production using perturbed Reynolds stress matrix */
+    
+    if (using_uq){
+      pw = PerturbedStrainMag * PerturbedStrainMag - TWO3*zeta*diverg;
```

Comment:
I'll be honest, I don't find the TWO3 thing any more readable than 2.0/3.0.

Reply:
I had just used it to be consistent because I saw it elsewhere in the code. Personally I find it even less readable than 2.0/3.0

Reply:
We have 3 votes for 2.0/3.0 already, I sense a replace all coming.

Reply:
Make it one vote more :)